src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/SelectionPropertyMetadataMapper.php

```diff
@@ -0,0 +1,114 @@
+    {
+        $mandatory = $propertyMetadata->isRequired();
+        $minMaxValue = static::getValidatedMinMaxValue($propertyMetadata);
+
+        return new ArrayMetadata(
+            $propertyMetadata->getName(),
+            $mandatory,
+            new SchemaMetadata([], [
+                new SchemaMetadata([], [], [], 'string'),
+                new SchemaMetadata([], [], [], 'number'),
+            ]),
+            $minMaxValue->min,
+            $minMaxValue->max,
+            true
+        );
+    }
+
+    /**
+     * Result object has following structure:
+     * <code>
+     * ?int $min - value of min parameter
+     * ?int $max - value of max parameter
+     * </code>.
+     *
+     * @throws \InvalidArgumentException
+     */
+    public static function getValidatedMinMaxValue(
+        ContentPropertyMetadata $propertyMetadata,
+        string $minParamName = 'min',
+        string $maxParamName = 'max'
+    ): object {
+        $mandatory = $propertyMetadata->isRequired();
+
+        $min = $propertyMetadata->getParameter($minParamName)['value'] ?? null;
+        Assert::nullOrIntegerish($min, \sprintf(
+            'Parameter "%s" of property "%s" needs to be either null or of type int',
+            $minParamName,
+            $propertyMetadata->getName()
+        ));
+
+        if (null !== $min) {
+            $min = (int) $min;
+        }
+
+        Assert::nullOrGreaterThanEq($min, 0, \sprintf(
+            'Parameter "%s" of property "%s" needs to be greater than or equal "0"',
+            $minParamName,
+            $propertyMetadata->getName()
+        ));
+
+        if ($mandatory) {
+            if (null === $min) {
+                $min = 1;
+            }
+
+            Assert::greaterThanEq($min, 1, \sprintf(
+                'Because property "%s" is mandatory, parameter "%s" needs to be greater than or equal "1"',
+                $propertyMetadata->getName(),
+                $minParamName
+            ));
+        }
+
+        $max = $propertyMetadata->getParameter($maxParamName)['value'] ?? null;
+        Assert::nullOrIntegerish($max, \sprintf(
+            'Parameter "%s" of property "%s" needs to be either null or of type int',
+            $maxParamName,
+            $propertyMetadata->getName()
+        ));
+
+        if (null !== $max) {
+            $max = (int) $max;
+        }
+
+        Assert::nullOrGreaterThanEq($max, 1, \sprintf(
+            'Parameter "%s" of property "%s" needs to be greater than or equal "1"',
+            $maxParamName,
+            $propertyMetadata->getName()
+        ));
+
+        if (null !== $min) {
+            Assert::nullOrGreaterThanEq($max, $min, \sprintf(
+                'Because parameter "%1$s" of property "%2$s" has value "%4$d", parameter "%3$s" needs to be greater than or equal "%4$d"',
+                $minParamName,
+                $propertyMetadata->getName(),
+                $maxParamName,
+                $min
+            ));
+        }
```

Comment:
I just checked, and `webmozart/assert` does not seem to be a direct dependency of Sulu, so we can't really use it here... Unless we add it as dependency, but I guess we would have to decide as a team if we want that.

Reply:
Since it's required by our dependencies either, I'm up for adding it to the `composer.json`.

Reply:
Since it also seems to be a popular package with multiple maintainers, I think it is a safe choice. We just have to make sure that everybody knows it is available, so that we also use it in other situations :slightly_smiling_face: /cc @sulu/core-team

Reply:
is okay for me 🙂

Reply:
Required it now, if someone has an objection, please share it here